### PR TITLE
`no-empty-file`: Added option to allow comments

### PR DESCRIPTION
**Pull request by AekoArray**
Hello, this is my first experience in open source. Could you see the implementation?

Fix https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2218

Allow only comments in files

```
"unicorn/no-empty-file": {
    "allow": [
        "comments"
	]
}
```

### CODE REVIEW

**Comment by sindresorhus** [comment]
Thanks for contributing. There are linting issues that needs to be resolved.

**Comment by AekoArray** [comment]
Hello! Can you please tell me how to fix linting issues? I tried running the command to update snapshots
`npm run test:js -- test/no-empty-file.mjs -u`
but it didn't help. Are there any other ways to fix them?

**Comment by fisker** [comment]
You need Node.js 20 to update snapshots  https://github.com/sindresorhus/eslint-plugin-unicorn/blob/231529aee793251f30416b93dc3d5f9bfa27ee47/.github/workflows/main.yml#L44

**Comment by AekoArray** [comment]
> You need Node.js 20 to update snapshots
> 
> https://github.com/sindresorhus/eslint-plugin-unicorn/blob/231529aee793251f30416b93dc3d5f9bfa27ee47/.github/workflows/main.yml#L44-L45

Thank you! This helped solve the linting problem. Could you please help with "Integration Test 3". Another PR has the same problem with this test. 
Also on the `main` branch I have the same issue with a "Integration Test 3"

**Comment by AekoArray** [comment]
@fisker Hello, could you check this PR again please?

File: rules/no-empty-file.js [hunk 1]
```diff
@@ -15,9 +15,17 @@ const isTripleSlashDirective = node =>
 const hasTripeSlashDirectives = comments =>
 	comments.some(currentNode => isTripleSlashDirective(currentNode));
 
+const isProgramFileEmpty = node => node.type === 'Program' && node.body.length === 0;
+
+const isAllowOnlyCommentsFile = (option, node) => option.allow.includes('comments') && isProgramFileEmpty(node);
```
Comment (fisker) [inline rules/no-empty-file.js:20]: I have difficult to understand this, shouldn't it be something like `allowComments && hasComment`?

Comment (fisker) [reply rules/no-empty-file.js:20]: I feel `/* comment */ {}` won't pass, but it should.

**Comment by sindresorhus** [comment]
@AekoArray Bump

File: rules/no-empty-file.js (hunk 1 above)
Comment (AekoArray) [reply rules/no-empty-file.js:20]: Thanks for review! I'll check it soon.

Comment (AekoArray) [reply rules/no-empty-file.js]: Done

**Review by fisker — APPROVED** [review]

File: docs/rules/no-empty-file.md [hunk 2]
```diff
@@ -68,3 +68,26 @@ const x = 0;
...
+
+```js
+"unicorn/no-empty-file": {
+    "allow": [
+        "comments"
+	]
+}
+```
+
+If you set it to `comments`, you can only store files with comments
```
Comment (sindresorhus) [inline docs/rules/no-empty-file.md:89]: This sentence could be made more clear.

File: docs/rules/no-empty-file.md [hunk 3]
```diff
@@ -68,3 +68,26 @@ const x = 0;
...
+
+Type: `Array<string>`\
+Default: `[]`
+
+You can set the `allow` option like this:
+
+```js
+"unicorn/no-empty-file": {
+    "allow": [
+        "comments"
```
Comment (sindresorhus) [inline docs/rules/no-empty-file.md:84]: Tab-indentation

**Review by sindresorhus — CHANGES_REQUESTED** [review]

**Comment by fisker** [comment]
Maybe the option should be `allowComments: true`?

**Comment by sindresorhus** [comment]
> Maybe the option should be allowComments: true?

Yeah, I was thinking that too 👍

**Comment by fisker** [comment]
Can you remove `.idea/*` files?

File: docs/rules/no-empty-file.md [hunk 4]
```diff
@@ -68,3 +68,24 @@ const x = 0;
 	const x = 0;
 }
 ```
+
+## Options
+
+### allow
```
Comment (fisker) [inline docs/rules/no-empty-file.md:74]:
```suggestion
### allowComments
```

Comment (AekoArray) [reply docs/rules/no-empty-file.md:74]: Done

**Comment by sindresorhus** [comment]
Bump :)

**Pull request closed without merging**